docker-compose.yml

```diff
@@ -0,0 +1,288 @@
+              "elasticsearch.service.local.environment": "true",
+              "elasticsearch.service.user": "elastic",
+              "elasticsearch.service.password": "changeme"
+            }'
+
+  messageapi:
+    container_name: messageapi
+    build:
+      context: .
+      args:
+        - API_DIR=/tmp/src/messageapi
+    env_file:
+      - .env.docker
+    environment:
+      - SPRING_MAIL_HOST=mailhog
+      - SPRING_MAIL_PORT=1025
+      - SPRING_MAIL_USERNAME=""
+      - SPRING_MAIL_PROPERTIES_MAIL_SMTP_AUTH=false
+      - SPRING_MAIL_PROPERTIES_MAIL_SMTP_STARTTLS_ENABLE=false
+    volumes:
+      - "${HOME}/.m2:/root/.m2"
+    ports:
+      - '8181:8080'
+    networks:
+      - common
+    depends_on:
+      - elasticsearch
+
+  mockproductmanagement:
+    container_name: mockproductmanagement
+    build:
+      context: .
+      args:
+        - API_DIR=/tmp/src/mockproductmanagement
+    volumes:
+      - "${HOME}/.m2:/root/.m2"
+    env_file:
+      - .env.docker
+    ports:
+      - '8182:8080'
+    networks:
+      - common
+    depends_on:
+      - elasticsearch
+    environment:
+      SPRING_APPLICATION_JSON: '{
+              "elasticsearch.service.url": "elasticsearch:9200",
+              "elasticsearch.service.local.environment": "true",
+              "elasticsearch.service.user": "elastic",
+              "elasticsearch.service.password": "changeme"
+            }'
+
+  orderapi:
+    container_name: orderapi
+    build:
+      context: .
+      args:
+        - API_DIR=/tmp/src/orderapi
+    volumes:
+      - "${HOME}/.m2:/root/.m2"
+    env_file:
+      - .env.docker
+    ports:
+      - '8183:8080'
+    networks:
+      - common
+    depends_on:
+      - elasticsearch
+    environment:
+      SPRING_APPLICATION_JSON: '{
+              "elasticsearch.service.url": "elasticsearch:9200",
+              "elasticsearch.service.local.environment": "true",
+              "elasticsearch.service.user": "elastic",
+              "elasticsearch.service.password": "changeme"
+            }'
+
+  paymentapi:
+    container_name: paymentapi
+    build:
+      context: .
+      args:
+        - API_DIR=/tmp/src/paymentapi
+    volumes:
+      - "${HOME}/.m2:/root/.m2"
+    env_file:
+      - .env.docker
+    ports:
+      - '8184:8080'
+    networks:
+      - common
+    depends_on:
+      - elasticsearch
+    environment:
+      SPRING_APPLICATION_JSON: '{
+              "elasticsearch.service.url": "elasticsearch:9200",
+              "elasticsearch.service.local.environment": "true",
+              "elasticsearch.service.user": "elastic",
+              "elasticsearch.service.password": "changeme"
+            }'
+
+  priceapi:
+    container_name: priceapi
+    build:
+      context: .
+      args:
+        - API_DIR=/tmp/src/priceapi
+    volumes:
+      - "${HOME}/.m2:/root/.m2"
+      - "./infra/priceapi/src/main/resources/application.properties:/tmp/src/priceapi/src/main/resources/application.properties"
```

Comment:
Only some of the backend api containers have properties files mapped as volumes. Should be consistent between all backend apis?